mempool/src/executor.rs

```diff
@@ -113,16 +123,37 @@ impl<N: Network, T: Topic + Unpin + Sync> Future for MempoolExecutor<N, T> {
                     TxPriority::Medium,
                 );
 
-                let acceptance = match verify_tx_ret {
-                    Ok(_) => MsgAcceptance::Accept,
-                    // Reject the message if signature verification fails or transaction is invalid
-                    // for current validation window
-                    Err(VerifyErr::InvalidTransaction(_)) => MsgAcceptance::Reject,
-                    Err(VerifyErr::AlreadyIncluded) => MsgAcceptance::Reject,
-                    Err(_) => MsgAcceptance::Ignore,
-                };
-
-                network.validate_message::<T>(pubsub_id, acceptance);
+                match pubsub_or_peer_id {
+                    PubsubIdOrPeerId::PeerId(peer_id) => match verify_tx_ret {
+                        Ok(_) => (),
+                        Err(VerifyErr::InvalidAccount(_))
+                        | Err(VerifyErr::InvalidBlockNumber)
+                        | Err(VerifyErr::InvalidTransaction(_)) => {
+                            if network.has_peer(peer_id) {
+                                warn!(
+                                    %peer_id,
+                                    "Banning peer because it responded with an invalid mempool transaction while syncing"
+                                );
+                                network
+                                    .disconnect_peer(peer_id, CloseReason::MaliciousPeer)
+                                    .await;
+                            }
+                        }
+                        Err(_) => (),
+                    },
+                    PubsubIdOrPeerId::PubsubId(pubsub_id) => {
+                        let acceptance = match verify_tx_ret {
+                            Ok(_) => MsgAcceptance::Accept,
+                            // Reject the message if signature verification fails or transaction is invalid
+                            // for current validation window
+                            Err(VerifyErr::InvalidTransaction(_)) => MsgAcceptance::Reject,
+                            Err(VerifyErr::AlreadyIncluded) => MsgAcceptance::Reject,
+                            Err(_) => MsgAcceptance::Ignore,
+                        };
+
+                        network.validate_message::<T>(pubsub_id, acceptance);
+                    }
```

Comment:
Can we somehow unify these branches so that we don't have duplication on what errors we consider bad?